node/node.go

```diff
@@ -393,10 +393,10 @@ func (node *AlgorandFullNode) Start() error {
 // Capabilities returns the node's capabilities for advertising to other nodes.
 func (node *AlgorandFullNode) Capabilities() []p2p.Capability {
 	var caps []p2p.Capability
-	if node.config.Archival {
+	if node.config.Archival && node.config.IsGossipServer() {
```

Comment:
Why does `isGossipServer()` have to return true to be archival? Same question for storesCatchpoints..

Reply:
IsGossipServer checks for NetAddress. Yes, a node needs to be a server to be able to accept block and ledger requests.